### ISSUE TITLE
Can not reproduce the results in CA.

### ISSUE THREAD

**Issue by FeiGSSS** [nokpil/AgentNet#2]
Hi,
Thanks for your contribution!  By running the code you provided,  I'm trying to reproduce the nice results of CA reported in your paper.  But there are several difficulties.

By default, the training process of CA is extremely confusing. The loss will first decrease and the attention scores will concentrate on the expected locality. But it is not a stable status, the loss will then vibrate to a new status where the attention seems to focus on partial-local-structure. And the summation of attention scores decreases to less than 1.

**Comment by FeiGSSS** [comment]
When I change the `shuffle` of `dataloader` to be all `True` , the loss will not vibrate and the ACC will retain >=99. However, the attention score will disappear (summations-->0) while the ACC still >=99.
1. Epoch 10, ACC>=99
<img width="364" alt="image" src="https://user-images.githubusercontent.com/29545851/155057597-ea92e4c5-c139-4063-9fb7-f5ac7522cb18.png">
2. Epoch 23, ACC>=99
<img width="364" alt="image" src="https://user-images.githubusercontent.com/29545851/155057724-31482aef-726d-4248-916a-fd09047f613b.png">

**Comment by nokpil** [comment]
Hi, 

Thanks for your attention. First of all, train_loader shuffle should be 'True' (obviously), I'll fix it.

The attention score in this paper is not a normalized one like a conventional 'attention' in ML practices (i.e., not softmax-ed among all other agents). Each attention score is output from a sigmoid function, thus may have any value between 0 to 1, (hence its sum can have any value between 0~N). Also, in theory, the exact scale of attention is not important in AgentNet, since the following decoder can scale up the signal whatever it likes. I guess the attention structure is still present, but not shown because now the maximum attention value becomes like 1e-2, not ~1.   (Check it by removing vlim for your imshow and plot it again). Another way to check that the attention structure is still present is by replacing the attention matrix with a zero-valued one and performing the forward pass. The difference in the result will prove that the attention score is still important (Different from being total zero), just scaled down.

But (nearly) vanishing attention is a somewhat interesting phenomenon. In my attempts, I've never encountered such a drastic decrease in overall attention value. Since you've reached the desired point at 10 epoch, (much faster than my paper has shown, 200 epoch), maybe you used more data (than the original version), and AgentNet somewhat adapted to that setting and comfortably reduce its overall scale (with more data, it can distinguish the relevant one and irrelevant one more precisely). 

Again, thanks for your attention. Feel free to leave a question and report!

**Comment by FeiGSSS** [comment]
> 

Thanks for your quick reply.  I removed the 0-1 limitation of `imshow`, and as you expected, the attention structure still holds while the absolute values decrease to near `1e-2`.
<img width="390" alt="image" src="https://user-images.githubusercontent.com/29545851/155068107-8c1ecd18-9dc0-453b-829e-b5ebe9aad2bb.png">
